pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -125,33 +133,39 @@ Future<Result, Consumer> MultiTopicsConsumerImpl::subscribeOneTopicAsync(const s
     }
 
     // subscribe for each partition, when all partitions completed, complete promise
-    lookupServicePtr_->getPartitionMetadataAsync(topicName).addListener(std::bind(
-        &MultiTopicsConsumerImpl::subscribeTopicPartitions, shared_from_this(), std::placeholders::_1,
-        std::placeholders::_2, topicName, subscriptionName_, conf_, topicPromise));
+    Lock lock(mutex_);
+    auto entry = topicsPartitions_.find(topic);
+    if (entry == topicsPartitions_.end()) {
+        lock.unlock();
+        lookupServicePtr_->getPartitionMetadataAsync(topicName).addListener(
+            [this, topicName, topicPromise](const Result result, const LookupDataResultPtr lookupDataResult) {
```

Comment:
```suggestion
            [this, topicName, topicPromise](Result result, const LookupDataResultPtr& lookupDataResult) {
```

The listener's function signature is `std::function<Result, const T&>`.